lenthall/src/main/scala/lenthall/collections/Table.scala

```diff
@@ -0,0 +1,24 @@
+package lenthall.collections
+
+object Table {
+  import lenthall.collections.EnhancedCollections._
+
+  /**
+    * Data structure to represent a table (with rows, columns and values)
+    * Really just an alias for Map[R, Map[C, V]]
+    * @tparam R type of the row
+    * @tparam C type of the column
+    * @tparam V type of the value
+    */
+  type Table[R, C, V] = Map[R, Map[C, V]]
```

Comment:
because `Map[(R, C), V]` can't give you "all the (C, V)s for an R" easily, which is one of the main reasons for this (give me all the shards for this node)

Reply:
FWIW I ran the `ExecutionStoreBenchmark` on 29_hotfix and on this branch and got a ~5 time speed up
29:
```
::Benchmark ExecutionStore.runnableCalls::
cores: 8
hostname: wm0e9-683
name: Java HotSpot(TM) 64-Bit Server VM
osArch: x86_64
osName: Mac OS X
vendor: Oracle Corporation
version: 25.51-b03
Parameters(size -> 10000): 32.85794863888889
Parameters(size -> 20000): 67.75936533333334
Parameters(size -> 30000): 101.32616099999998
Parameters(size -> 40000): 136.29065163888887
Parameters(size -> 50000): 165.96757452777777
Parameters(size -> 60000): 211.2232175555556
Parameters(size -> 70000): 242.80586366666662
Parameters(size -> 80000): 276.8677186111111
Parameters(size -> 90000): 309.74371752777773
Parameters(size -> 100000): 347.79028586111116
```

Table:
```
::Benchmark ExecutionStore.runnableCalls::
cores: 8
hostname: wm0e9-683
name: Java HotSpot(TM) 64-Bit Server VM
osArch: x86_64
osName: Mac OS X
vendor: Oracle Corporation
version: 25.51-b03
Parameters(size -> 10000): 7.858199194444448
Parameters(size -> 20000): 16.204352833333335
Parameters(size -> 30000): 26.52700425
Parameters(size -> 40000): 36.46423988888889
Parameters(size -> 50000): 31.638888416666678
Parameters(size -> 60000): 58.2120958611111
Parameters(size -> 70000): 59.49086905555555
Parameters(size -> 80000): 60.948874388888875
Parameters(size -> 90000): 78.167269
Parameters(size -> 100000): 96.13690763888893
```

Reply:
That would complicate looking up all shards for a scatter